app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,30 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+  end
 
-  def show; end
+  def show
+    @post = User.find(params[:user_id]).posts.find(params[:id])
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(title: post_params[:title], text: post_params[:text], author_id: current_user[:id],
+                     commentsCounter: 0, likesCounter: 0)
+
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ ] Great job creating the create method, but I couldn't create a post on behalf of a current user and I got an error. Kindly fix this issue and test the app before submitting it for another review. Please note that due to the same error, I failed to create a comment or a like because a post should exist before a comment or a like.

![new-post](https://github.com/Jolak5/AI_update/assets/53124548/f14ed893-3443-41e7-9337-41fe04bf5a9a)